src/material/core/focus_indicator/_focus-indicator.scss

```diff
@@ -0,0 +1,54 @@
+@import '../theming/theming';
+
+@mixin mat-focus-indicator() {
+  // Focus indicator base styles.
+  .mat-focus-indicator::before {
+    border-radius: 4px;
+    border-width: 2px;
+    border-style: solid;
+    box-sizing: border-box;
+    position: absolute;
+    pointer-events: none;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+  }
+
+  // Focus indicator offsets.
+  .mat-focus-indicator {
+    &.mat-button-base::before,
+    &.mat-card::before,
+    &.mat-calendar-body-cell::before,
+    &.mat-sort-header-button::before {
+      margin: -4px;
```

Comment:
Would it make sense to add classes like `.mat-focus-indicator-inset` and `.mat-focus-indicator-outset` and apply that to the appropriate elements instead of encoding specific elements here?

Reply:
Seems reasonable to me. If we do so, I'll probably change the offset for mat-tab-label/mat-tab-link to be `margin: 4px` instead of what I have now just for consistency.

Reply:
I ended up not doing this because different elements need slightly different default insets/outsets.

Reply:
That's reasonable for now- can you add a comment that explains this?